pkg/kubectl/cmd/set/set_probe.go

```diff
@@ -0,0 +1,452 @@
+
+	  # Set an HTTP readiness probe for port 8080 and path /healthz over HTTP on the pod IP
+	  kubectl set probe deploy/webapp --readiness --get-url=http://:8080/healthz
+
+	  # Set an HTTP readiness probe over HTTPS on 127.0.0.1 for a hostNetwork pod
+	  kubectl set probe deploy/router --readiness --get-url=https://127.0.0.1:1936/stats
+
+	  # Set only the initial-delay-seconds field on all deployments
+	  kubectl set probe deploy --all --readiness --initial-delay-seconds=30`)
+)
+
+type ProbeOptions struct {
+	Out io.Writer
+	Err io.Writer
+	resource.FilenameOptions
+	ContainerSelector      string
+	Selector               string
+	All                    bool
+	Output                 string
+	Builder                *resource.Builder
+	Infos                  []*resource.Info
+	Encoder                runtime.Encoder
+	ShortOutput            bool
+	DryRun                 bool
+	Record                 bool
+	Cmd                    *cobra.Command
+	ChangeCause            string
+	Mapper                 meta.RESTMapper
+	PrintObject            func(cmd *cobra.Command, isLocal bool, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error
+	UpdatePodSpecForObject func(obj runtime.Object, fn func(*v1.PodSpec) error) (bool, error)
+	Resources              []string
+
+	Readiness bool
+	Liveness  bool
+	Remove    bool
+	Local     bool
+
+	OpenTCPSocket string
+	HTTPGet       string
+	Command       []string
+
+	FlagSet       func(string) bool
+	HTTPGetAction *v1.HTTPGetAction
+
+	// Length of time before health checking is activated.  In seconds.
+	InitialDelaySeconds *int
+	// Length of time before health checking times out.  In seconds.
+	TimeoutSeconds *int
+	// How often (in seconds) to perform the probe.
+	PeriodSeconds *int
+	// Minimum consecutive successes for the probe to be considered successful after having failed.
+	// Must be 1 for liveness.
+	SuccessThreshold *int
+	// Minimum consecutive failures for the probe to be considered failed after having succeeded.
+	FailureThreshold *int
+}
+
+// NewCmdProbe implements the set probe command
+func NewCmdProbe(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ProbeOptions{
+		Out: out,
+		Err: errOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "probe RESOURCE/NAME --readiness|--liveness [options] (--get-url=URL|--open-tcp=PORT|-- CMD)",
+		Short:   "Update a probe on a pod template",
+		Long:    probeLong,
+		Example: probeExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run(f))
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "identifying the resource to get from a server."
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", "*", "The names of containers in the selected pod templates to change - may use wildcards")
```

Comment:
I'd prefer to use `field-selector` here with `metadata.name=abc` instead of `ContainerSelector`.

Reply:
`-c` safe with log command... and feild selector can get slice now？